fastify-session-authentication/authentication.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+const fastify = require('fastify')
+const fastifySession = require('fastify-session')
+const fastifyCookie = require('fastify-cookie')
+const fastifyFormbody = require('fastify-formbody')
+
+const { loginPage, defaultPage } = require('./html')
+
+// create a fastify instance
+const app = fastify()
```

Comment:
I would prefer if we exposed a factory that returned `app` without calling `app.listen()` and then call that only if this is the startup file.

Reply:
I would suggest using this approach with fastify-cli
https://github.com/fastify/example/blob/aa633714867da5edc029700145a5a7505be2b880/validation-messages/package.json#L7

so you write less code and more testable

Reply:
I changed it to export a plugin and using `fastify-cli` to start the example.